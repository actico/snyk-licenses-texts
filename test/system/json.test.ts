import { exec } from 'child_process';
import { sep } from 'path';
const main = './dist/index.js'.replace(/\//g, sep);
const ORG_ID = process.env.TEST_ORG_ID as string;
describe('`snyk-licenses-report json <...>`', () => {
  it('Shows error when missing --orgPublicId', async (done) => {
    exec(`node ${main} json`, (err, stdout) => {
      expect(stdout).toBe("");
      expect(err.message.trim()).toMatchSnapshot();
      done();
    });
  });
  it('Generated JSON data with correct --orgPublicId', async (done) => {
    exec(`DEBUG=snyk-license* node ${main} json --orgPublicId=${ORG_ID}`, (err, stdout) => {
      expect(err).toBeNull();
      console.log({err, stdout, ORG_ID})
      expect(stdout.trim()).toMatch("BSD-2-Clause");
      done();
    });
  }, 30000);
});
