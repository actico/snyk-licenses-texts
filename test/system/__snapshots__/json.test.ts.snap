// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-licenses-report json <...>\` Shows error when missing --orgPublicId 1`] = `
"Command failed: node ./dist/index.js json
index.js json

Generate org licenses & dependencies data in JSON format

Options:
  --version      Show version number                                   [boolean]
  --help         Show help                                             [boolean]
  --orgPublicId                                                       [required]

Missing required argument: orgPublicId"
`;
